docs/docsite/rst/network/user_guide/cli_parsing.rst

```diff
@@ -0,0 +1,728 @@
+.. _cli_parsing:
+
+*****************************************
+Parsing Semi-structured Text with Ansible
+*****************************************
+
+The `ansible.netcommon <https://galaxy.ansible.com/ansible/netcommon>`_ collection version 1.2.0 or later  includes the :ref:`cli_parse <ansible_collections.ansible.netcommon.cli_parse_module>` module that can run commands and parse the output. You can use the ``cli_parse`` module on a device, host, or platform that only supports a command-line interface and the commands issued return semi-structured text.
+
+.. contents::
+   :local:
+
+
+Understanding the CLI parser
+=============================
+
+The ``cli_parse`` module can either run a CLI command on a device and return a parsed result or can simply parse any text document. The ``cli_parse`` module includes cli_parser plugins to interface with a variety of parsing engines.
+
+Why parse the text?
+--------------------
+
+Parsing non-structured data such as network configurations into structured data allows programmatic use of the data from that device. You can pull information from a network device and update a CMDB in one playbook. Use cases include automated troubleshooting, creating dynamic documentation, updating IPAM (IP address management) tools and so on. You may prefer to do this with Ansible natively to take advantage of native Ansible constructs such as:
```

Comment:
The page title refers to "semi-structured" text, but here we talk about transforming "non-structured" data into "structured" data. Are these two different use cases? If not, I'd suggest picking a term and using it in both places.

Reply:
fixed